firebase-firestore/src/main/java/com/google/firebase/firestore/local/LruGarbageCollector.java

```diff
@@ -15,16 +15,82 @@
 package com.google.firebase.firestore.local;
 
 import android.util.SparseArray;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
 import com.google.firebase.firestore.core.ListenSequence;
+import com.google.firebase.firestore.util.Logger;
 import java.util.Comparator;
 import java.util.PriorityQueue;
 
 /** Implements the steps for LRU garbage collection. */
-class LruGarbageCollector {
+public class LruGarbageCollector {
+  public static class Params {
+    private static final long COLLECTION_DISABLED = FirebaseFirestoreSettings.CACHE_SIZE_UNLIMITED;
+    private static final long DEFAULT_CACHE_SIZE_BYTES = 100 * 1024 * 1024; // 100mb
+
+    public static Params Default() {
+      return new Params(DEFAULT_CACHE_SIZE_BYTES, 10, 1000);
+    }
+
+    public static Params Disabled() {
+      return new Params(COLLECTION_DISABLED, 0, 0);
+    }
+
+    public static Params WithCacheSize(long cacheSizeBytes) {
```

Comment:
Maybe rename to `WithCacheSizeBytes`, since that makes the calling code much more descriptive: `WithCacheSize(100)` versus `WithCacheSizeBytes(100)`.